case_conversion/case_parse.py

```diff
@@ -7,75 +7,199 @@
+            case_type = 'camel'
+        elif pascal_case:
+            case_type = 'pascal'
+        else:
+            case_type = 'mixed'
+
+    return case_type
+
+# Check a run of words represented by the range [s, i]. Should
+# return last index of new word groups.
+
+
+def _advanced_acronym_detection(s, i, words, acronyms):
+    # Combine each letter into single string.
+    acstr = ''.join(words[s:i])
+
+    # List of ranges representing found acronyms.
+    range_list = []
+    # Set of remaining letters.
+    not_range = set(range(len(acstr)))
+
+    # Search for each acronym in acstr.
+    for acronym in acronyms:
+        # TODO: Sanitize acronyms to include only letters.
+        rac = regex.compile(unicode(acronym))
+
+        # Loop until all instances of the acronym are found,
+        # instead of just the first.
+        n = 0
+        while True:
+            m = rac.search(acstr, n)
+            if not m:
+                break
+
+            a, b = m.start(), m.end()
+            n = b
+
+            # Make sure found acronym doesn't overlap with others.
+            ok = True
+            for r in range_list:
+                if a < r[1] and b > r[0]:
+                    ok = False
+                    break
+
+            if ok:
+                range_list.append((a, b))
+                for j in xrange(a, b):
+                    not_range.remove(j)
+
+    # Add remaining letters as ranges.
+    for nr in not_range:
+        range_list.append((nr, nr + 1))
+
+    # No ranges will overlap, so it's safe to sort by lower bound,
+    # which sort() will do by default.
+    range_list.sort()
+
+    # Remove original letters in word list.
+    for _ in xrange(s, i):
+        del words[s]
+
+    # Replace them with new word grouping.
+    for j in xrange(len(range_list)):
+        r = range_list[j]
+        words.insert(s + j, acstr[r[0]:r[1]])
+
+    return s + len(range_list) - 1
+
+# Fallback to simple acronym detection.
+
+
+def _simple_acronym_detection(s, i, words, *args):
+    # Combine each letter into a single string.
+    acronym = ''.join(words[s:i])
+
+    # Remove original letters in word list.
+    for _ in xrange(s, i):
+        del words[s]
+
+    # Replace them with new word grouping.
+    words.insert(s, ''.join(acronym))
+
+    return s
+
+
+class InvalidAcronymError(Exception):
+    """Raise when acronym fails validation."""
+
+    def __init__(self, acronym):
+        super(InvalidAcronymError, self).__init__()
+        m = "Case Conversion: acronym '{}' is invalid."
+        print(m.format(acronym))
+
+
+def _sanitize_acronyms(unsafe_acronyms):
+    # Sanitize acronyms list by discarding invalid acronyms and
+    # normalizing valid ones to upper-case.
+    valid_acronym = regex.compile(u'^[\p{Ll}\p{Lu}\p{Nd}]+$')
+    acronyms = []
+    for a in unsafe_acronyms:
+        if valid_acronym.match(a):
+            acronyms.append(a.upper())
+        else:
+            raise InvalidAcronymError(a)
```

Comment:
Note: In case of invalid acronyms a custom exception will now be raised. I feel this makes more sense in the context of this module that can be used in a multitude of configurations compared to the original plugin it stems from. It is more in accordance with points 10/11 of Zen of Python.

Reply:
Yeah, it looks clean